integration/linux/build/Dockerfile-centos.template

```
@@ -9,46 +9,73 @@ ENV LANG en_US.UTF-8
 
 ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib"
 
+%%IF VARIANT=7%%
+ENV GCC_VERSION 11
+%%ENDIF%%
+
 # rpm on centos 7 iterates over all fds up to the limit, which is
 # extremely slow. Force it to be small with ulimit.
 #
 # We need to disable the mirrorlist and explicitly set up a
 # baseurl. Switch to the archive.kernel.org mirror while we are at it.
-RUN ulimit -n 1024 \
+RUN set -ex \
+	&& ulimit -n 1024 \
+	&& sed -i 's/enabled=1/enabled=0/g' /etc/yum/pluginconf.d/fastestmirror.conf \
     && sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-* \
     && sed -i 's|# \?baseurl=http://mirror.centos.org|baseurl=https://archive.kernel.org/centos-vault/|g' /etc/yum.repos.d/CentOS-* \
-    && yum update -y
+    && yum update -y \
+%%IF VARIANT=7%%
+    && yum install -y centos-release-scl epel-release \
+    && sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-SCLo* \
+    && sed -i "s|# \?baseurl=http://mirror.centos.org|baseurl=https://archive.kernel.org/centos-vault/|g" /etc/yum.repos.d/CentOS-SCLo* \
+    && yum update -y \
+%%ENDIF%%
+	&& echo "repo setup done"
 
 # rpm on centos 7 iterates over all fds up to the limit, which is
 # extremely slow. Force it to be small.
 RUN ulimit -n 1024 \
     && yum install -y \
-    wget gcc gcc-c++ make patch zlib-devel openssl-devel sqlite-devel bzip2-devel \
+    wget gcc gcc-c++ make patch zlib-devel sqlite-devel bzip2-devel \
+	openssl-devel \
```

If needed, `IFNOT` is also available:

```suggestion
%%IFNOT VARIANT=7%%
	openssl-devel \
%%ENDIF%%
```

---

That package is also required